torchci/clickhouse_queries/lf_rollover_health/query.sql

```diff
@@ -1,144 +1,96 @@
-      WHERE 1 = 1
-        AND j.job_name = mj.job_name
-        -- AND STRPOS(j.name, mj.job_clean) > 0
-
-), success_stats AS (
+    normalized_jobs as j
+  WHERE
+    j.label LIKE 'lf%'
+),
+comparable_jobs AS (
+  SELECT
+    j.bucket,
+    j.started_at,
+    j.duration_min,
+    j.label,
+    j.job_name,
+    j.workflow_name,
+    j.conclusion
+  FROM
+    normalized_jobs as j
+  CROSS JOIN lf_jobs as lfj
+  WHERE
+    j.job_name = lfj.job_name
+),
+success_stats AS (
   SELECT
     bucket,
     count(*) as group_size,
     job_name,
     workflow_name,
     label,
-    IF(SUBSTR(label, 1, 3) = 'lf.', True, False ) as lf_fleet,
-    SUM(
-        CASE
-            WHEN conclusion = 'success' THEN 1
-            ELSE 0
-        END
-    ) * 100 / (COUNT_IF(conclusion != 'cancelled') + 1) as success_rate, -- plus one is to handle divide by zero errors
-    SUM(
-        CASE
-            WHEN conclusion = 'failure' THEN 1
-            ELSE 0
-        END
-    ) * 100 / (COUNT_IF(conclusion != 'cancelled') + 1) as failure_rate,
-    SUM(
-        CASE
-            WHEN conclusion = 'cancelled' THEN 1
-            ELSE 0
-        END
-    ) * 100 / COUNT(*) as cancelled_rate, -- cancelled rate is calculated over all jobs
-    SUM(
-        CASE
-            WHEN conclusion = 'success' THEN 1
-            ELSE 0
-        END
-    ) as success_count,
-    SUM(
-        CASE
-            WHEN conclusion = 'failure' THEN 1
-            ELSE 0
-        END
-    ) as failure_count,
-    SUM(
-        CASE
-            WHEN conclusion = 'cancelled' THEN 1
-            ELSE 0
-        END
-    ) as cancelled_count,
-    COUNT(*) as total_count,
-    SUM(
-        CASE
-            WHEN conclusion = 'success' THEN duration_min
-            ELSE 0
-        END
-    ) / COUNT(*) as success_avg_duration,
-    SUM(
-        CASE
-            WHEN conclusion = 'failure' THEN duration_min
-            ELSE 0
-        END
-    ) / COUNT(*) as failure_avg_duration,
-    SUM(
-        CASE
-            WHEN conclusion = 'cancelled' THEN duration_min
-            ELSE 0
-        END
-    ) / COUNT(*) as cancelled_avg_duration,
-
+    if(startsWith(label, 'lf.'), 1, 0 ) as lf_fleet,
+    sum(case when conclusion = 'success' then 1 else 0 end) * 100 / (countIf(conclusion != 'cancelled') + 1) as success_rate, -- plus one is to handle divide by zero errors
+    sum(case when conclusion = 'failure' then 1 else 0 end) * 100 / (countIf(conclusion != 'cancelled') + 1) as failure_rate,
+    sum(case when conclusion = 'cancelled' then 1 else 0 end) * 100 / COUNT() as cancelled_rate,
+    sum(case when conclusion = 'success' then duration_min else 0 end) / countIf(conclusion = 'success') as success_avg_duration,
+    sum(case when conclusion = 'failure' then duration_min else 0 end) / countIf(conclusion = 'failure') as failure_avg_duration,
+    sum(case when conclusion = 'cancelled' then duration_min else 0 end) / countIf(conclusion = 'cancelled') as cancelled_avg_duration
```

Comment:
CH seems to have a sumIf operator if you want to simplify this, possibly also an avgIf function

Reply:
ooh, nice